solr/solrj/src/test/org/apache/solr/client/solrj/GetByIdTest.java

```diff
@@ -30,15 +35,19 @@ public class GetByIdTest extends EmbeddedSolrServerTestBase {
 
   @BeforeClass
   public static void beforeClass() throws Exception {
-    initCore();
+    solrClientTestRule
+        .build()
+        .setSolrHome(Paths.get(SolrJettyTestBase.legacyExampleCollection1SolrHome()))
+        .init();
   }
 
   @Before
   @Override
   public void setUp() throws Exception {
     super.setUp();
-    getSolrClient().deleteByQuery("*:*");
-    getSolrClient()
+    solrClientTestRule.getSolrClient().deleteByQuery("*:*");
+    solrClientTestRule
+        .getSolrClient()
```

Comment:
Here and in many places in this file, you no longer call getSolrClient(), instead, more verbosely, you go to the solrClientTestRule first.  Is getSolrClient is still defined in EmbeddedSolrServerTestBase which is fine to continue to use?

Reply:
Should I define getSolrClient() in EmbeddedSolrServerTestBase to return the client created in the test rule or I should revert to how it was defined initially?

Reply:
> Should I define getSolrClient() in EmbeddedSolrServerTestBase to return the client created in the test rule

Yes.